### PR TITLE
docs: adds instructions for collecting and sending native histograms with otel collector

### PR DESCRIPTION
**Pull request by tacole02**
#### What this PR does

This PR updates https://grafana.com/docs/mimir/next/send/native-histograms/ with instructions for collecting and sending native histograms with the OTel Collector.

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir-squad/issues/2131)](https://github.com/grafana/mimir-squad/issues/2131

#### Checklist

- [ ] Tests updated.
- [X] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: docs/sources/mimir/send/native-histograms/_index.md [hunk 1]
```diff
@@ -184,6 +184,33 @@ Use the latest version of [Grafana Alloy](https://grafana.com/docs/alloy/<ALLOY_
...
+   ```
+
+   For more information, refer to [Registering Views](https://opentelemetry.io/docs/languages/go/instrumentation/#registering-views) in the OpenTelemetry SDK documentation for Go.
+
+### Instrument the OpenTelemetry Collector using Java
+
+Use the OpenTelemetry SDK version 1.35.0 or later.
+
+1. Set up the collector to handle your metrics data. This includes setting up your resources, meter provider, meter, instruments, and views. Refer to [Metrics](https://opentelemetry.io/docs/languages/java/instrumentation/#metrics) in the OpenTelemetry SDK documentation for Java.
+1. To aggregate a histogram instrument as an exponential histogram, include the following view:
```
Comment (tacole02) [inline docs/sources/mimir/send/native-histograms/_index.md:212]: @krajorama could you please provide an example implementation for Java? Thanks!

Comment (krajorama) [reply docs/sources/mimir/send/native-histograms/_index.md:212]: I've tried to set it up, but I don't have enough Java experience at this point to get it to work, I suggest we stick to Go for now. Maybe someone can contribute the Java parts once we have the structure and everything figured out for Go.

Comment (tacole02) [reply docs/sources/mimir/send/native-histograms/_index.md:212]: Sounds good! I'll make the docs Go-specific for the time being.

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+1. To aggregate a histogram instrument as an exponential histogram, include the following view:
+
+   ```
+   Aggregation: metric.AggregationBase2ExponentialHistogram{
+			MaxSize:  160,
+			MaxScale: 20,
+		}
+   ```
+
+   For more information, refer to [Registering Views](https://opentelemetry.io/docs/languages/go/instrumentation/#registering-views) in the OpenTelemetry SDK documentation for Go.
```
Comment (zhehao-grafana) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:34]: I am wondering whether we should add a link regarding the parameter settings: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#base2-exponential-bucket-histogram-aggregation

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
+description: Learn how to collect and send exponential histograms with the OpenTelemetry Collector
+menuTitle: OpenTelemetry exponential histograms
+title: Collect and send exponential histograms with the OpenTelemetry Collector
+weight: 100
+---
+
+# Collect and send exponential histograms with the OpenTelemetry Collector
+
+OpenTelemetry [exponential histograms](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#exponentialhistogram) are compatible with Prometheus native histograms. The only significant difference is that exponential histograms store the `min` and `max`  observation values explicitly, but native histograms don't, which means you need to estimate the minimum and maximum by using the 0.0 and 1.0 quantile respectively.
```
Comment (tacole02) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:10]: Do you mean you need to estimate the min and max values for native histograms? If that's the case, can we rephrase this to indicate that for exponential histograms, you don't need to estimate these values, since they are explicitly stored?

Since this topic if about exponential histograms, I think it makes more sense to have that be the focus.

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  - OpenTelemetry
+  - instrumentation
+menuTitle: OpenTelemetry exponential histograms
+title: Send exponential histograms to Mimir
+weight: 200
+---
+
+# Send exponential histograms to Mimir
+
+You can collect and send exponential histograms to Mimir with the OpenTelemetry Collector. OpenTelemetry [exponential histograms](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#exponentialhistogram) are compatible with Prometheus native histograms. The key difference is that exponential histograms store the `min` and `max`  observation values explicitly, whereas native histograms don't. This means that for exponential histograms, you don't need to estimate these values using the 0.0 and 1.0 quantiles.
```
Comment (tacole02) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:15]: @krajorama , I massaged this a little bit to keep the topic's focus on exponential histograms. lmk if this works.

Comment (tacole02) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:15]: @krajorama , I massaged this a little bit to keep the topic's focus on exponential histograms. lmk if this works.

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+   ```
+   Aggregation: metric.AggregationBase2ExponentialHistogram{
+			MaxSize:  160,
+			MaxScale: 20,
+		}
+   ```
+
+   For more information, refer to [Registering Views](https://opentelemetry.io/docs/languages/go/instrumentation/#registering-views) in the OpenTelemetry SDK documentation for Go.
+
+## Migrate from classic histograms
```
Comment (tacole02) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:43]: [@krajorama](https://github.com/krajorama) do the migration instructions at https://grafana.com/docs/mimir/next/send/native-histograms/#migrate-from-classic-histograms not apply here? If that's the case, how can people migrate?

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+## Bucket boundary calculation
```
Comment (tacole02) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:45]: @krajorama same question as above.

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md (hunk 2 above)
Comment (krajorama) [reply docs/sources/mimir/send/otel-exponential-histograms/_index.md:41]: nice, that's a good idea I think

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md (hunk 5 above)
Comment (krajorama) [reply docs/sources/mimir/send/otel-exponential-histograms/_index.md:43]: The migration starts differently, but from the dashboards (step 5) is the same.

In OTEL the explicit bucket and exponential bucket histograms cannot easily co-exist as the name of the metric would clash. So you start from an existing histogram that uses explicit buckets and add a new [view](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view) where you assign a new *name* as well as exponential aggregation to the histogram. This will result in a new metric with the new name and exponential buckets. For example I have in my test code a histogram called `request_latency` and I can add a new view that will create `request_latency_exp` metric with exponential buckets.
```
v := sdkmetric.NewView(sdkmetric.Instrument{
		Name: "request_latency",
		Kind: sdkmetric.InstrumentKindHistogram,
	}, sdkmetric.Stream{
		Name: "request_latency_exp",
		Aggregation: sdkmetric.AggregationBase2ExponentialHistogram{MaxSize: 160, NoMinMax: true, MaxScale: 20},
		//Aggregation: sdkmetric.DefaultAggregationSelector(sdkmetric.InstrumentKindHistogram),
	})
```

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md (hunk 6 above)
Comment (krajorama) [reply docs/sources/mimir/send/otel-exponential-histograms/_index.md:45]: Almost the same, except the bucket offsets are shifted by one, so in native histogram you have
image
but in otel you have
otel-bound

This isn't terribly interesting for end-users :) See also https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4a5e6db494acb2d56617266828f8db03957d3960/pkg/translator/prometheusremotewrite/histograms.go#L107

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 7]
```diff
@@ -38,8 +38,77 @@ Use the OpenTelemetry SDK version 1.17.0 or later.
...
+1. It's important to keep scraping both explicit bucket and exponential histograms for at least the period of the longest range in your recording rules and alerts, plus one day. This is the minimum amount of time, but it's recommended to keep scraping both data types until you can verify the new rules and alerts.
+
+   For example, if you have an alert that calculates the rate of requests, such as `sum(rate(http_request_duration_seconds[7d]))`, this query looks at the data from the last seven days plus the Prometheus [lookback period](https://prometheus.io/docs/prometheus/latest/querying/basics/#staleness). When you start sending exponential histograms, the data isn't there for the entire seven days, and therefore, the results might be unreliable for alerting.
+
+1. After configuring exponential histogram collection, choose one of the following ways to stop collecting explicit bucket histograms.
+
+   - Remove the custom bucket definition, `Buckets`/`classicUpperBounds`, from the instrumentation. In Java, also use the `nativeOnly()` option. Refer to the examples in [Instrument application with Prometheus client libraries](#instrument-application-with-prometheus-client-libraries).
```
Comment (krajorama) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:108]: This "Remove the custom bucket definition" is Prometheus specific, I would not go into details, just mention to remove explicit bucket definition and/or view that exposes the explicit buckets.

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 8]
```diff
@@ -38,8 +38,77 @@ Use the OpenTelemetry SDK version 1.17.0 or later.
...
+   - Drop the explicit bucket histogram series with [Prometheus relabeling](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) or [Grafana Alloy prometheus.relabel](https://grafana.com/docs/alloy/<ALLOY_VERSION>/reference/components/prometheus/prometheus.relabel) at the time of scraping.
```
Comment (krajorama) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:109]: I haven't tried it but in the Otel collector you have something like [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md) that should be possible to utilize for the same purpose. Let's not mix Prometheus or Alloy here for now.

**Review by krajorama — COMMENTED** [review]
we're getting there

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md (hunk 6 above)
Comment (tacole02) [reply docs/sources/mimir/send/otel-exponential-histograms/_index.md]: In that case, is this something users need to be aware of at all? Or is it just something that happens in the background?

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+The OpenTelemetry Collector supports collecting exponential histograms and other compatible data formats, including native histograms and Datadog sketches, through its receivers and sending them through its exporters.
+
+{{< admonition type="note" >}}
+The availability of different receivers and exporters depends on your OpenTelemetry Collector [distribution](https://opentelemetry.io/docs/concepts/distributions/).
+{{< /admonition >}}
+
+You can use the OpenTelemetry protocol (OTLP) to send exponential histograms to Grafana Mimir in their existing format, or you can use the Prometheus remote write protocol to send them as Prometheus native histograms. Refer to [Send native histograms to Mimir](https://grafana.com/docs/mimir/<MIMIR_VERSION>/send/native-histograms/).
```
Comment (krajorama) [inline docs/sources/mimir/send/otel-exponential-histograms/_index.md:23]: We did not document sending with OTEL collector, so let's remove the reference. Also I realized when looking into this that there's two OTLP exporters and we only support the one over HTTP protocol.

```suggestion
You can use the OpenTelemetry protocol (OTLP) over HTTP to send exponential histograms to Grafana Mimir in their existing format, or you can use the Prometheus remote write protocol to send them as Prometheus native histograms.
```

We could refer to the the OTLP HTTP exporter and Promtheus remote write exporter documentation:
- https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/otlphttpexporter/README.md
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/README.md

File: docs/sources/mimir/send/otel-exponential-histograms/_index.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+	   	Kind: sdkmetric.InstrumentKindHistogram,
+	   }, sdkmetric.Stream{
+		   Name: "request_latency_exp",
+		   Aggregation: sdkmetric.AggregationBase2ExponentialHistogram{MaxSize: 160, NoMinMax: true, MaxScale: 20},
+	   })
+   ```
+
+   For more information about creating a view, refer to [View](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view) in the OpenTelemetry Metrics SDK.
+
+1. Modify dashboards to use the exponential histogram metrics. Refer to [Visualize native histograms](https://grafana.com/docs/mimir/<MIMIR_VERSION>/visualize/native-histograms/) for more information.
```
[the remark this answers is not in the document]
Comment (tacole02) [reply docs/sources/mimir/send/otel-exponential-histograms/_index.md:64]: I can look into this idea for a future PR, especially when this topic goes GA and will have wider readership. We would still need to substitute all of the classic/native details for explicit bucket/exponential, so I'm not sure it would be worth the effort.

**Comment by tacole02** [comment]
@krajorama WDYT? Can we merge? 🚢 🚢

**Review by krajorama — APPROVED** [review]
LGTM
